kfs-core/src/main/java/org/kuali/kfs/fp/batch/ProcurementCardAutoApproveDocumentsStep.java

```diff
@@ -22,10 +22,13 @@
 
 import org.kuali.kfs.fp.batch.service.ProcurementCardCreateDocumentService;
 import org.kuali.kfs.sys.batch.AbstractStep;
+import org.kuali.rice.coreservice.framework.parameter.ParameterConstants.NAMESPACE;
 
 /**
  * This step will call a service method to autoapprove any documents that have been in route passed a certain number of days.
  */
+//This annotation is needed to lookup parameters correctly
+@NAMESPACE( namespace = "KFS-PCARD" )
```

Comment:
The sleep is needed, in this case, to make sure all just routed documents have been successfully routed before we check to see which documents need to be auto-approved. Our Procurement Card batch process is more complicated and has several more steps than the delivered KFS 6.0 batch job.